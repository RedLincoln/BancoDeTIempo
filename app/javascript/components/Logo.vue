<template>
  <v-img :max-width="maxWidth" :src="src"> </v-img>
</template>

<script>
import logoDark from "images/logo_dark.png";
import logo from "images/logo.png";

export default {
  props: {
    maxWidth: {
      type: String,
      default: "150",
    },
    dark: {
      type: Boolean,
      default: false,
    },
  },
  computed: {
    src() {
      return this.dark ? logoDark : logo;
    },
  },
};
</script>

<style>
</style>