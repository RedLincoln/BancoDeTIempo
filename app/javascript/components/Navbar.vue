<template>
  <v-app-bar color="deep-purple" dense dark app>
    <v-toolbar-title>Banco de Tiempo</v-toolbar-title>
    <v-spacer></v-spacer>

    <v-btn
      :to="{ name: 'login' }"
      color="primary"
      class="mr-2"
      data-testid="login-btn"
      >Iniciar Sesi√≥n</v-btn
    >
    <v-btn :to="{ name: 'sign_up' }" color="secondary" data-testid="sign_up-btn"
      >Crear Cuenta</v-btn
    >
  </v-app-bar>
</template>
