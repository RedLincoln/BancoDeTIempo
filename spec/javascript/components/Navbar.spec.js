import Vuetify from "vuetify";
import Vuex from "vuex";

import Navbar from "../../../app/javascript/components/Navbar.vue";
import Logo from "../../../app/javascript/components/Logo.vue";

import { createLocalVue, mount } from "@vue/test-utils";

describe("Navbar.vue", () => {
  const localVue = createLocalVue();
  localVue.component("Logo", Logo);
  let vuetify;
  let wrapper;
  let state;
  let store;

  beforeEach(() => {
    jest.resetAllMocks();
    localVue.use(Vuex);

    state = {
      user_name: "Michael",
      loggedIn: false,
    };
    store = new Vuex.Store({
      modules: {
        session: {
          namespaced: true,
          state,
        },
      },
    });

    vuetify = new Vuetify();
    wrapper = mount(Navbar, {
      localVue,
      vuetify,
      store,
      stubs: ["router-link"],
    });
  });

  it("renders properly", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
