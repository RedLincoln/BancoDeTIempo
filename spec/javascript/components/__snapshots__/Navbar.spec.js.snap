// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar.vue renders properly 1`] = `
<header class="v-sheet theme--dark v-toolbar v-toolbar--dense v-app-bar v-app-bar--fixed deep-purple" style="height: 48px; margin-top: 0px; transform: translateY(0px); left: 0px; right: 0px;">
  <div class="v-toolbar__content" style="height: 48px;">
    <div class="v-toolbar__title">Banco de Tiempo</div>
    <div class="spacer"></div>
    <router-link-stub class="mr-2 v-btn v-btn--contained v-btn--router theme--dark v-size--default primary" data-testid="login-btn"><span class="v-btn__content">Iniciar Sesi√≥n</span></router-link-stub>
    <router-link-stub class="v-btn v-btn--contained v-btn--router theme--dark v-size--default secondary" data-testid="sign_up-btn"><span class="v-btn__content">Crear Cuenta</span></router-link-stub>
  </div>
</header>
`;
